### PR TITLE
DMD-917 - Implement kbc llm init command

### PR DESCRIPTION
**Pull request by vojtabiberle**
## Release Notes
Added new `kbc llm init` command that initializes a local directory for LLM context generation. This is the first step toward the `kbc llm export` feature that will generate AI-optimized project context for use with AI assistants.

## Plans for customer communication
None.

## Impact analysis
- Adds new `llm` command group to the CLI
- Refactors `AskAllowedBranches` dialog to be reusable across `sync init` and `llm init` commands
- No changes to existing `sync init` behavior (tests updated to use new function signature)

## Change type
Feature

## Justification
This implements PR 1 of the LLM Twin Format Export RFC (DMD-917). The `kbc llm init` command is a prerequisite for the upcoming `kbc llm export` command.

Linear issue: https://linear.app/keboola/issue/DMD-917/pr-1-complete-kbc-llm-init-branch-jt-llm-init

## Deployment
Merge & automatic deploy.

## Rollback plan
Revert of this PR.

## Post release support plan
None.

---

## Human Review Checklist

**Key areas to review:**
1. The refactored `AskAllowedBranches` function moved from `sync/init/dialog.go` to `dialog/allowed_branches.go` - verify the behavior is preserved for existing `sync init` command
2. New `SelectBranches` method added to dialog package - verify it handles edge cases correctly
3. No E2E tests included for `kbc llm init` - the RFC mentions tests should be included

---

**Link to Devin run:** https://app.devin.ai/sessions/eecace21ad294aaabf3504f33fff40dc
**Requested by:** vojtech.biberle@keboola.com (@vojtabiberle)

### CODE REVIEW

File: internal/pkg/service/cli/cmd/sync/init/dialog.go
Comment (Matovidlo) [inline internal/pkg/service/cli/cmd/sync/init/dialog.go]: Why did we need to rework this?

File: pkg/lib/operation/llm/init/operation.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	createManifest "github.com/keboola/keboola-as-code/pkg/lib/operation/project/local/manifest/create"
+	createMetaDir "github.com/keboola/keboola-as-code/pkg/lib/operation/project/local/metadir/create"
+	genWorkflows "github.com/keboola/keboola-as-code/pkg/lib/operation/project/local/workflows/generate"
+)
+
+type Options struct {
+	ManifestOptions createManifest.Options
+	Workflows       genWorkflows.Options
```
[the remark this answers is not in the document]
Comment (vojtabiberle) [reply pkg/lib/operation/llm/init/operation.go]: removed

[the remark this answers is not in the document]
Comment (vojtabiberle) [reply pkg/lib/operation/llm/init/operation.go]: removed
